cloudfoundry/data_source_cf_user_provided_service_instance.go

```diff
@@ -0,0 +1,59 @@
+package cloudfoundry
+
+import (
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2/constant"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-cloudfoundry/cloudfoundry/managers"
+)
+
+func dataSourceUserProvidedServiceInstance() *schema.Resource {
+
+	return &schema.Resource{
+
+		Read: dataSourceUserProvidedServiceInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"space": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": &schema.Schema{
```

Comment:
Maybe simply retrieving tags is not enough here.
User probably want all informations from provided service (credentials included)

Reply:
I am not sure you can retrieve more meaningful information with the client library that is used. The returned instance is of type ServiceInstance which does not have the credentials info.

Reply:
I don't remember why i did that sadly but as you can see here: https://github.com/cloudfoundry-community/cloudfoundry-cli/blame/v0.0.2-complete-api/api/cloudcontroller/ccv2/user_provided_service_instance.go#L87 you should use `GetUserProvServiceInstances` instead of `GetUserProvidedServiceInstances`.
Looks like insane but there must be a good reason (or a simple bad copy and paste at the beginning).
Using `GetUserProvServiceInstances` will give you all informations as documented here: http://apidocs.cloudfoundry.org/12.31.0/user_provided_service_instances/list_all_user_provided_service_instances.html